### PR TITLE
Do not display "English" as default language on resources

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#3957 *A course clearly in spanish displaying "English" as its language, because the course did not specify a language and "English" was our default.*

#### What's this PR do?
This PR changes our Learning Resource displays (card, drawer) to not show "English" as the default course language. If a run has no language specified, then no language row is displayed in course info section.

#### How should this be manually tested?
1. **no language specified**:  At http://od.odl.local:8063/infinite or http://od.odl.local:8063/infinite/search, click on a card title. Most resources have no language specified, so clicking any of them should be fine.
    - In network tab, check the API response for the resource details. In "runs"  property, check that no language is specified.
    - Check that no language is displayed in the learning resource drawer
2. **Language is displayed if exists on resource runs:** Only `mitx` resources have languages defined, and most of the time it's English.
    - Find some MITx courses that have language specified (see below), and check that it displays correctly in the learning resource drawer. 

To find MITx course that have a language specified, you can run `docker compose run --rm web ./manage.py shell` and then run
```python
from course_catalog.models import LearningResourceRun

languages = ["tr", "ko", "en-us", "es-es", None]
base_url = "http://cac.od.odl.local:8063/infinite"
for lang in languages:
    try:
        filter_kwrags = {
            "language": lang,
        } if lang is not None else { "language__isnull": True }
        course = next(
            run for run in
            LearningResourceRun.objects.filter(platform='mitx', **filter_kwrags, published=True)
            if getattr(run.content_object, "published")
        ).content_object
        print(f"{lang} course:")
        print(f"{base_url}?resource_id={course.id}&resource_type=course")
    except StopIteration:
        print(f"Could not find a published course with language {lang}")
        continue
```

Which should print you the URL of some resources with non-null langauges.

#### Screenshots (if appropriate)

Screenshot 2023-04-25 at 4 38 31 PM
Screenshot 2023-04-25 at 4 38 19 PM

### CODE REVIEW

File: frontends/ol-search-ui/src/util.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {
 } from "./interfaces"
 import React, { useState, useEffect } from "react"
 import { capitalize, emptyOrNil } from "ol-util"
-import LocaleCode from "locale-code"
+import ISO6391 from "iso-639-1"
```
Comment (ChristopherChudzicki) [inline frontends/ol-search-ui/src/util.tsx:11]: We had been using [locale-code](https://www.npmjs.com/package/locale-code) for language names. Swap to [iso-639-1](https://www.npmjs.com/package/iso-639-1?activeTab=readme). These packages are by the same author & `language-code depends` on `iso-639-1`, so we already had it as a dependency. 

The difference is:
- `iso-639-1` expects a language code... 'en', 'fr', etc
- `locale-code` expects a locale-code, `en-US`, `en-GB`, `zh-CN`, etc

**The reason to swap is:**
- we had been appending a country code 'US' to **all** language codes. That was arbitrary and confusing, but was required to `locale-code`.

File: frontends/ol-search-ui/src/util.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  it("Returns null if input is null or undefined", () => {
+    expect(languageName(undefined)).toBe(null)
+    expect(languageName(null)).toBe(null)
+  })
+
+  it("Returns null if input is not a 2-digit ISO-639-1 code", () => {
+    expect(languageName("zz")).toBe(null)
+    expect(languageName("")).toBe(null)
+    expect(languageName("eng")).toBe(null)
+  })
```
Comment (ChristopherChudzicki) [inline frontends/ol-search-ui/src/util.test.ts:33]: This whole test suite is new, but these two tests are the changed behavior. Previously they would have returned "en-US".

**Review by mbertrand — CHANGES_REQUESTED** [review]
Works for  `../infinite/search`, but English is still showing up as the default under `../learn/search` via [frontends/open_discussions/src/components/ExpandedLearningResourceDisplay.js](https://github.com/mitodl/open-discussions/blob/f0d300bd6b480836f291d8ff5f860bfc6faeaec8/frontends/open-discussions/src/components/ExpandedLearningResourceDisplay.js#L331)

**Comment by ChristopherChudzicki** [comment]
> Works for ../infinite/search, but English is still showing up as the default under ../learn/search via [frontends/open_discussions/src/components/ExpandedLearningResourceDisplay.js](https://github.com/mitodl/open-discussions/blob/f0d300bd6b480836f291d8ff5f860bfc6faeaec8/frontends/open-discussions/src/components/ExpandedLearningResourceDisplay.js#L331)

That was deliberate, though in this case the bug fix is super simple so probably worthfixing both cases.

If a less trivial issue comes up, maybe worth more of a discussion.

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**